### ISSUE TITLE
Build fails when using both route groups and global-error.tsx

### ISSUE THREAD

**Issue by navFooh** [vercel/next.js#47748]
### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

```bash
Operating System:
      Platform: win32
      Arch: x64
      Version: Windows 10 Home
    Binaries:
      Node: 16.16.0
      npm: N/A
      Yarn: N/A
      pnpm: N/A
    Relevant packages:
      next: 13.2.5-canary.22
      eslint-config-next: N/A
      react: 18.2.0
      react-dom: 18.2.0
```

### Which area(s) of Next.js are affected? (leave empty if unsure)

App directory (appDir: true)

### Link to the code that reproduces this issue

https://github.com/navFooh/route-groups

### To Reproduce

- Create a new layout in `app/(group)/layout.tsx`
- Create a new page in `app/(group)/route/page.tsx`
- Create `global-error.tsx`
- Run `npm run build`

### Describe the Bug

The build fails with the following error:

```bash
Generating static pages (0/4)SyntaxError: Unexpected token u in JSON at position 0
    at JSON.parse (<anonymous>)
    at U (C:\dev\ttt\route-groups\node_modules\next\dist\compiled\react-server-dom-webpack\cjs\react-server-dom-webpack-client.edge.production.min.js:22:53)
    at c (C:\dev\ttt\route-groups\node_modules\next\dist\compiled\react-server-dom-webpack\cjs\react-server-dom-webpack-client.edge.production.min.js:24:307)

Error occurred prerendering page "/". Read more: https://nextjs.org/docs/messages/prerender-error
SyntaxError: Unexpected token u in JSON at position 0
    at JSON.parse (<anonymous>)
    at U (C:\dev\ttt\route-groups\node_modules\next\dist\compiled\react-server-dom-webpack\cjs\react-server-dom-webpack-client.edge.production.min.js:22:53)
    at c (C:\dev\ttt\route-groups\node_modules\next\dist\compiled\react-server-dom-webpack\cjs\react-server-dom-webpack-client.edge.production.min.js:24:307)
    
> Build error occurred
ExportError: Export encountered errors on following paths:
        /page: /
    at C:\dev\ttt\route-groups\node_modules\next\dist\export\index.js:469:19
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async Span.traceAsyncFn (C:\dev\ttt\route-groups\node_modules\next\dist\trace\trace.js:79:20)
    at async C:\dev\ttt\route-groups\node_modules\next\dist\build\index.js:1459:21
    at async Span.traceAsyncFn (C:\dev\ttt\route-groups\node_modules\next\dist\trace\trace.js:79:20)
    at async C:\dev\ttt\route-groups\node_modules\next\dist\build\index.js:1317:17
    at async Span.traceAsyncFn (C:\dev\ttt\route-groups\node_modules\next\dist\trace\trace.js:79:20)
    at async Object.build [as default] (C:\dev\ttt\route-groups\node_modules\next\dist\build\index.js:74:29) {
  type: 'ExportError'
}
```

### Expected Behavior

The build should succeed

**Comment by icazemier** [comment]
Got a similar error:


(`"next": "13.2.4",`)

```
SyntaxError: Unexpected token u in JSON at position 0
    at JSON.parse (<anonymous>)
    at T (/Volumes/Development/somethingsomething/node_modules/next/dist/compiled/react-server-dom-webpack/cjs/react-server-dom-webpack-client.edge.production.min.js:22:53)
    at c (/Volumes/Development/somethingsomething/node_modules/next/dist/compiled/react-server-dom-webpack/cjs/react-server-dom-webpack-client.edge.production.min.js:24:307)

```

(When omitting the `global-error.tsx` no error appears, seems solved)

One difference with your issue maybe, I get the error in runtime inside the browser output, not in the build itself.
Maybe because I got this in every page.tsx:

```
export const dynamic = 'force-dynamic';
export const dynamicParams = true;
```

**Comment by meslzy** [comment]
+1

**Comment by navFooh** [comment]
Fixed for me using next 13.3.1, so closing the issue
